src/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch/ElasticSearchOutput.cs

```diff
@@ -378,11 +382,77 @@ private async Task EnsureIndexExists(string indexName, ElasticClient esClient)
             }
         }
 
-        private string GetIndexName(ElasticSearchConnectionData connectionData)
+        protected internal string GetIndexName(ElasticSearchConnectionData connectionData)
+        {
+            string dateString = "";
+            try
+            {
+                dateString = ConvertIndexFormat(connectionData.Configuration.IndexFormat);
+            }
+            catch (Exception ex)
+            {
+                string errorMessage = $"{nameof(ElasticSearchOutput)}: indexFormat could not parse as date: {ex.Message}{Environment.NewLine}" +
+                                      $"Invalid IndexFormat value: {connectionData.Configuration.IndexFormat}{Environment.NewLine}" +
+                                      $"Falling back to default value: {ElasticSearchOutputConfiguration.DefaultIndexFormat}";
+                // Changing in connectionData.Configuration so we don't report this every log message
+                connectionData.Configuration.IndexFormat = ElasticSearchOutputConfiguration.DefaultIndexFormat;
+                healthReporter.ReportWarning(errorMessage);
+                dateString = string.Format($"{{0:{connectionData.Configuration.IndexFormat}}}", DateTimeOffset.UtcNow);
+            }
+
+            return connectionData.Configuration.IndexNamePrefix + dateString;
+        }
+
+        public Func<DateTimeOffset> Now { get; set; } = DefaultNow;
+        public static DateTimeOffset DefaultNow() { return DateTimeOffset.UtcNow; }
+
+        protected internal string ConvertIndexFormat(string dateString)
```

Comment:
I am wondering how fast this method is. .NET has become quite efficient these days, but still, this will be called for every batch of events, so it might be beneficial to optimize things a bit. A few suggestions below.